template.yaml

```diff
@@ -2296,25 +2295,51 @@ Resources:
       Environment:
         Variables:
           XML_BUCKET_NAME: !Ref ScopusXmlBucket
+          SCOPUS_QUEUE: !Ref ScopusImportQueue
       PackageType: Zip
       InlineCode: |
         import boto3, zipfile, os
         from io import BytesIO
+  
         def lambda_handler(event, context):
             xml_bucket = os.environ['XML_BUCKET_NAME']
+            scopus_queue = os.environ['SCOPUS_QUEUE']
             zip_bucket = event['Records'][0]['s3']['bucket']['name']
             zip_file = event['Records'][0]['s3']['object']['key']
-            client = boto3.client('s3')
+  
+            s3_client = boto3.client('s3')
             resource = boto3.resource('s3')
+            sqs_client = boto3.client('sqs')
+  
             zip_object = resource.Object(bucket_name=zip_bucket, key=zip_file)
             buffer = BytesIO(zip_object.get()['Body'].read())
             z = zipfile.ZipFile(buffer)
+  
             for fname in z.namelist():
-                client.put_object(
-                    Body=z.open(fname).read(),
-                    Bucket=zip_bucket if fname.endswith('.zip') else xml_bucket,
-                    Key=f'unzipped/{fname}' if fname.endswith('.zip') else fname
-                )
+                file_content = z.open(fname).read()
+                if fname.endswith('.zip'):
+                    s3_client.put_object(
+                        Body=file_content,
+                        Bucket=zip_bucket,
+                        Key=f'unzipped/{fname}'
+                    )
+                else:
+                    s3_client.put_object(
+                        Body=file_content,
+                        Bucket=xml_bucket,
+                        Key=fname
+                    )
+                    s3_uri = f's3://{xml_bucket}/{fname}'
+                    sqs_client.send_message(
+                        QueueUrl=scopus_queue,
+                        MessageBody=fname,
+                        MessageAttributes={
+                            'uri': {
+                                'DataType': 'String',
+                                'StringValue': s3_uri
+                            }
+                        }
+                    )
```

Comment:
I put some effort to write it in java, but is was not so straightforward and code was ugly, so I decided simply to extend existing inline python code :)